api/src/OpenTelemetry/SemConvStabilityOptIn.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module OpenTelemetry.SemConvStabilityOptIn (
+  getSemConvStabilityOptIn,
+  SemConvStabilityOptIn (Stable, Both, Old),
+) where
+
+import qualified Data.Text as T
+import System.Environment (lookupEnv)
+
+
+data SemConvStabilityOptIn = Stable | Both | Old deriving (Show, Eq)
+
+
+getSemConvStabilityOptIn :: IO SemConvStabilityOptIn
```

Comment:
I split it into two functions, but I am thinking about keeping the tests the same. I like that the current tests make sure that the program is reading the env var correctly, and the fact that `getSemConvStabilityOptIn` is trivial means that it is still a good test for `parseSemConvStabilityOptIn`. It might be better to test the pure function though, not sure.

Reply:
your tests are pretty nice, compact and legible, but one thing you could do with the pure function is reduce the number of env var tests to two (set/unset), and replace the rest with tests of the pure function that could be one-liners like `parseSemConvStabiltyOptIn (Just "http") `shouldBe` Stable`. if the setup was more onerous than `setEnv "blah"` I might feel more strongly about suggesting that approach, but as it stands I think it's just an aesthetic choice.

Reply:
I updated the tests to use a looping structure like Max suggested. Does that satisfy you're aesthetic preference?

Reply:
that's good too :)